docs/api/index.md

```diff
@@ -2592,23 +2592,114 @@ Vitest provides utility functions to help you out through it's **vi** helper. Yo
+
+  This also means that you cannot use any variables inside the factory that are defined outside of the factory.
+
+  If you need to use variables inside the factory, try [`vi.doMock`](#vi-domock). It works the same way, but isn't hoisted. Beware that it only mocks subsequent imports.
+  :::
+
+  ::: warning
+  If you are mocking a module with default export, you will need to provide a `default` key within the returned factory function object. This is an ES modules specific caveat, therefore `jest` documentation may differ as `jest` uses CommonJS modules. For example,
 
   ```ts
-  vi.mock('path', () => {
+  vi.mock('./path/to/module.js', () => {
     return {
       default: { myDefaultKey: vi.fn() },
       namedExport: vi.fn(),
       // etc...
     }
   })
   ```
+  :::
+
+  If there is a `__mocks__` folder alongside a file that you are mocking, and the factory is not provided, Vitest will try to find a file with the same name in `__mocks__` subfolder and use it as an actual module. If you are mocking a dependency, Vitest will try to find a `__mocks__` folder in the [root](/config/#root) of the project (default is `process.cwd()`).
+
+  For example, you have this file structure:
+
+  ```
+  - __mocks__
+    - axios.ts
+  - src
+    __mocks__
+      - increment.ts
+    - increment.ts
+  - tests
+    - increment.test.ts
+  ```
+
+  If you call `vi.mock` in a test file without a factory provided, it will find file in `__mocks__` folder to use as a module:
+
+  ```ts
+  // increment.test.ts
+  import { vi } from 'vitest'
+  // axios is a default export from `__mocks__/axios.ts`
+  import axios from 'axios'
+  // increment is a named export from `src/__mocks__/increment.ts`
+  import { increment } from '../increment.ts'
+
+  vi.mock('axios')
+  vi.mock('../increment.ts')
+
+  axios.get(`/apples/${increment(1)}`)
+  ```
+
+  ::: warning
+  Beware that if you don't call `vi.mock`, modules **are not** mocked automatically.
+  :::
 
-  - If `__mocks__` folder with file of the same name exist, all imports will return its exports. For example, `vi.mock('axios')` with `<root>/__mocks__/axios.ts` folder will return everything exported from `axios.ts`.
-  - If there is no `__mocks__` folder or a file with the same name inside, will call original module and mock it. (For the rules applied, see [algorithm](/guide/mocking#automocking-algorithm).)
+  If there is no `__mocks__` folder or a factory provided, Vitest will import original module and automock all its exports. For the rules applied, see [algorithm](/guide/mocking#automocking-algorithm).
+
+### vi.doMock
+
+- **Type**: `(path: string, factory?: () => unknown) => void`
+
+  The same as [`vi.mock`](#vi-mock), but is not hoisted to top of the file, so you can reference variables in the global file scope. The next import of the module will be mocked. This will not mock modules that were imported before this was called.
```

Comment:
```suggestion
  The same as [`vi.mock`](#vi-mock), but it's not hoisted at the top of the file, so you can reference variables in the global file scope. The next import of the module will be mocked. This will not mock modules that were imported before this was called.
```